### ISSUE TITLE
Comment: Heading styles need to be unique

### ISSUE THREAD

**Issue by MartinKorz** [Nathan-Yorio/obsidian-autotocgen-hack#1]
Thanks for a great script! I just set it up for my vault and style and my test documents work well.

One detail I noticed is that the different heading specifications in the `recipe` variable such as
```
[[heading]]
# Heading 5
level = 5
greedy = true
font.name = "Inter-SemiBold"
font.size = 8.414400100708008'
```
need to be unique. So if one's obsidian theme uses the same font and font-size for different heading levels, one would need to add e.g. the `font.color` variable to make things work.

Now that I had this realization it is kind of obvious, but it took me a few minutes until it clicked.

**Comment by Y0RI0** [comment]
> Thanks for a great script! I just set it up for my vault and style and my test documents work well.
> 
> One detail I noticed is that the different heading specifications in the `recipe` variable such as
> 
> ```
> [[heading]]
> # Heading 5
> level = 5
> greedy = true
> font.name = "Inter-SemiBold"
> font.size = 8.414400100708008'
> ```
> 
> need to be unique. So if one's obsidian theme uses the same font and font-size for different heading levels, one would need to add e.g. the `font.color` variable to make things work.
> 
> Now that I had this realization it is kind of obvious, but it took me a few minutes until it clicked.

Yes, nice catch. This is a caveat of using this method. It may also produce issues for you if you frequently manually change the zoom level of text in obsidian. I'm not sure of a good workaround as this was a quick hack for me, but if you keep a single theme and don't change zoom you can get away with repeating the same recipe for other documents.
